### PR TITLE
METROL-103: update WebKitPlugin

### PR DESCRIPTION
**Pull request by MFransen69**
This is the same PR as created against the Master branch ((https://github.com/rdkcentral/rdkservices/pull/257) but now against the Sprint branch. It also includes the changes resulting from the latest discussion on the DiskCacheDir and WebKit responsiveness check code.

### CODE REVIEW

**Review by pwielders — DISMISSED** [review]

File: WebKitBrowser/WebKitImplementation.cpp [hunk 1]
```diff
@@ -730,30 +746,42 @@ static GSourceFuncs _handlerIntervention =
...
+            return Core::ERROR_NONE;
         }
 
-        virtual void SetHeaders(const string& headers) final
+        uint32_t Headers(const string& headers) override
         {
-            if (_context != nullptr) {
-                using SetHeadersData = std::tuple<WebKitImplementation*, string>;
-                auto* data = new SetHeadersData(this, headers);
+            _adminLock.Lock();
```
Comment (emutavchi) [inline WebKitBrowser/WebKitImplementation.cpp:773]: Please keep the headers setting through the SetHeadersData and remove extra locking _adminLock. This avoids deadlocks that we see when the requester tries to set multiple properties and webkit implementation tries to emit a notificatoin

File: WebKitBrowser/WebKitImplementation.cpp [hunk 2]
```diff
@@ -766,93 +794,90 @@ static GSourceFuncs _handlerIntervention =
...
+            return Core::ERROR_NONE;
         }
 
-        virtual void SetUserAgent(const string& useragent) final
+        uint32_t UserAgent(const string& ua) override
         {
-            if (_context == nullptr)
-                return;
+            _adminLock.Lock();
+            _config.UserAgent = ua;
```
Comment (emutavchi) [inline WebKitBrowser/WebKitImplementation.cpp:815]: Same here and all setters bellow

**Review by emutavchi — CHANGES_REQUESTED** [review]

File: WebKitBrowser/ResidentApp.config [hunk 3]
```diff
@@ -1,5 +1,4 @@
 set (autostart ${PLUGIN_RESIDENT_APP_AUTOSTART})
-set (resumed true)
```
[the remark this answers is not in the document]
Comment (MFransen69) [reply WebKitBrowser/ResidentApp.config:2]: Yes, no problem will change this.

File: WebKitBrowser/WebKitImplementation.cpp (hunk 1 above)
Comment (MFransen69) [reply WebKitBrowser/WebKitImplementation.cpp:773]: I see no harm in moving setting the member to inside the lambda so no problem (certainly if you mention it prevents deadlocks of course :) ), so I will change this

File: WebKitBrowser/WebKitImplementation.cpp (hunk 2 above)
Comment (MFransen69) [reply WebKitBrowser/WebKitImplementation.cpp:815]: Yes, will do

**Comment by MFransen69** [comment]
A new one against the sprint/2010 branch was created

**Pull request closed without merging**